sentinelhub/evalscript.py

```diff
@@ -0,0 +1,117 @@
+"""
+Module defining evalscript generation utilities
+"""
+from typing import Dict, List, Optional
+
+import numpy as np
+
+from .data_collections import DataCollection
+from .data_collections_bands import Band, Unit
+
+DTYPE_TO_SAMPLE_TYPE: Dict[type, str] = {
+    bool: "UINT8",
+    np.uint8: "UINT8",
+    np.uint16: "UINT16",
+    np.float32: "FLOAT32",
+}
+
+EVALSCRIPT_TEMPLATE = """
+    //VERSION=3
+
+    function setup() {{
+        return {{
+            input: [{{
+                bands: [{input_names}],
+                units: [{input_units}]
+            }}],
+            output: [{output_spec}]
+        }}
+    }}
+
+    function updateOutputMetadata(scenes, inputMetadata, outputMetadata) {{
+        outputMetadata.userData = {{
+            "norm_factor":  inputMetadata.normalizationFactor
+        }}
+    }}
```

Comment:
will this always be available? even when not using `DN`?

Reply:
Yes, the logic behind it is to always have it available, and the user uses this based on their specific needs, as in they should know what they're doing.

If the user doesn't specify the `userdata` in the response, then this doesn't get downloaded anyway.

Reply:
but does the `normalizationFactor` exist if one downloads with REFLECTANCE?

Reply:
Yes, because the info is taken from the metadata of the collection. I tried it for a couple of different collections as well, it's always set to some value and based on the docs always represents the factor in the formula for getting reflectances from DN.